### ISSUE TITLE
The result of softmax is not same with numpy/pytorch

### ISSUE THREAD

**Issue by FrankyTang** [NVIDIA/TensorRT#3077]
## Description

I tried to run softmax for a tensor with axis=1, but the result  laneline_class_out_softmax[0,:,0,0].sum() is not  equal to 1. like picture
<img width="967" alt="image" src="https://github.com/NVIDIA/TensorRT/assets/32587916/9a0d1cf7-421b-42ba-82ae-bc8f67e82f69">

source onnx in here .
[s.onnx.zip](https://github.com/NVIDIA/TensorRT/files/11814056/s.onnx.zip)

**Comment by FrankyTang** [comment]
I found it is truly difference. try/onnxruntime will convert tensor to [N,D] with axis. eg, tensor [1,10,96,160], axis=2, softmax will convert to [10, 15360], 15360=96*160.And than the sum of D dimension is equal to 1.
